secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/internationalpayments/InternationalPaymentsApiController.java

```diff
@@ -0,0 +1,278 @@
+        if (!isPaymentSubmission.isPresent()) {
+            return ResponseEntity.status(BAD_REQUEST).body("Payment submission '" + internationalPaymentId + "' can't be found");
+        }
+
+        FRInternationalPaymentSubmission frPaymentSubmission = isPaymentSubmission.get();
+        OBVersion apiVersion = VersionPathExtractor.getVersionFromPath(request);
+        if (!ResourceVersionValidator.isAccessToResourceAllowed(apiVersion, frPaymentSubmission.getObVersion())) {
+            return PaymentApiResponseUtil.resourceConflictResponse(frPaymentSubmission, apiVersion);
+        }
+        //get the consent
+        final InternationalPaymentConsent consent = consentStoreClient.getConsent(frPaymentSubmission.getConsentId(), apiClientId);
+        logger.debug("Got consent from store: {}", consent);
+
+        return ResponseEntity.ok(responseEntity(consent, frPaymentSubmission));
+    }
+
+    @Override
+    public ResponseEntity getInternationalPaymentsInternationalPaymentIdPaymentDetails(
+            String internationalPaymentId,
+            String authorization,
+            String xFapiAuthDate,
+            String xFapiCustomerIpAddress,
+            String xFapiInteractionId,
+            String xCustomerUserAgent,
+            String apiClientId,
+            HttpServletRequest request,
+            Principal principal) {
+
+        Optional<FRInternationalPaymentSubmission> isInternationalPaymentSubmission = paymentSubmissionRepository.findById(internationalPaymentId);
+        if (!isInternationalPaymentSubmission.isPresent()) {
+            return ResponseEntity.status(BAD_REQUEST).body("International payment submission '" + internationalPaymentId + "' can't be found");
+        }
+
+        FRInternationalPaymentSubmission frInternationalPaymentSubmission = isInternationalPaymentSubmission.get();
+        logger.debug("Found The International Payment '{}' to get details.", internationalPaymentId);
+        OBVersion apiVersion = VersionPathExtractor.getVersionFromPath(request);
+        if (!ResourceVersionValidator.isAccessToResourceAllowed(apiVersion, frInternationalPaymentSubmission.getObVersion())) {
+            return PaymentApiResponseUtil.resourceConflictResponse(frInternationalPaymentSubmission, apiVersion);
+        }
+        return ResponseEntity.ok(responseEntityDetails(frInternationalPaymentSubmission));
+    }
+
+    private OBWriteInternationalResponse5 responseEntity(
+            InternationalPaymentConsent consent,
+            FRInternationalPaymentSubmission frPaymentSubmission
+    ) {
+        FRWriteInternationalData data = frPaymentSubmission.getPayment().getData();
+        OBWriteInternationalResponse5Data responseData = new OBWriteInternationalResponse5Data();
+
+        final Optional<FRInternationalResponseDataRefund> refundAccountData = refundAccountService.getInternationalPaymentRefundData(
+                consent.getRequestObj().getData().getReadRefundAccount(),
```

Comment:
Trivial but you could make a var for `consent.getRequestObj().getData()` (and `getInitiation()`) to avoid wasted calls.

Reply:
You're right but I wanted to maintain the same pattern for both supported versions.
Doing that would mean I'll have to refactor, basically change this part on all the controllers across all versions.

Reply:
Would keep that in mind though.

Reply:
Thanks!